WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java

```diff
@@ -497,7 +494,18 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                                     : AnalyticsTracker.Stat.ME_GRAVATAR_GALLERY_PICKED;
                     AnalyticsTracker.track(stat);
                     Uri imageUri = Uri.parse(strMediaUri);
-                    startCropActivity(imageUri);
+                    if (!MediaUtils.isInMediaStore(imageUri)) {
+                        // Download the picture. See https://github.com/wordpress-mobile/WordPress-Android/issues/5818
+                        Uri downloadedUri = MediaUtils.downloadExternalMedia(getActivity(), imageUri);
```

Comment:
Never mind that. It does lock the main thread, but I thought it actually downloaded the media over the network which would've caused an ANR.